File: unimonpark/src/main/java/co/edu/unimonserrate/unimonpark/entity/Externo.java
package co.edu.unimonserrate.unimonpark.entity;

import java.time.LocalDateTime;

import co.edu.unimonserrate.unimonpark.enums.TipoDocumento;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "externos")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Externo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idExterno;

    @Enumerated(EnumType.STRING)
    @Column(name = "tipo_documento", nullable = false, length = 20)
    private TipoDocumento tipoDocumento;

    @Column(name = "numero_documento", nullable = false, unique = true, length = 30)
    private String numeroDocumento;

    @Column(nullable = false, length = 100)
    private String nombres;

    @Column(nullable = false, length = 100)
    private String apellidos;

    @Column(length = 20)
    private String telefono;

    @Column(length = 100)
    private String correo;

    @Column(length = 150)
    private String empresa;

    @Column(nullable = false)
    private Boolean activo = true;

    @Column(name = "fecha_creacion", nullable = false)
    private LocalDateTime fechaCreacion;
}

File: unimonpark/unimonpark-frontend/src/types/externo.ts
export type TipoDocumento = "CC" | "TI" | "CE" | "PASAPORTE" | "OTRO";

export interface Externo {
    idExterno: number;
    tipoDocumento: TipoDocumento;
    numeroDocumento: string;
    nombres: string;
    apellidos: string;
    telefono?: string | null;
    correo?: string | null;
    empresa?: string | null;
    activo: boolean;
    fechaCreacion?: string;
}

export type ExternoPayload = Omit<Externo, "idExterno" | "fechaCreacion">;